core/domain/services/registration/CopyRegistrationService.php

```diff
@@ -174,4 +176,45 @@ public function copyPaymentDetails(
         }
         return true;
     }
+
+
+    /**
+     * @param EE_Registration $target_registration
+     * @param EE_Registration $registration_to_copy
+     * @return bool
+     * @throws RuntimeException
+     * @throws UnexpectedEntityException
+     * @throws EE_Error
+     */
+    public function copyPromotionLineItems(
+        EE_Registration $target_registration,
+        EE_Registration $registration_to_copy
+    ) {
+        $transaction = $registration_to_copy->transaction();
+        if ($transaction instanceof \EE_Transaction) {
+            $total_line_item = $transaction->total_line_item();
+            $old_event_line_item = EEH_Line_Item::get_event_line_item_for_ticket($total_line_item, $registration_to_copy->ticket());
+            $new_event_line_item = EEH_Line_Item::get_event_line_item_for_ticket($total_line_item, $target_registration->ticket());
+            $promo_line_items = EEH_Line_Item::get_line_items_of_object_type($old_event_line_item, \EEM_Line_Item::OBJ_TYPE_PROMOTION);
```

Comment:
do we need to cancel these too? or was that already done (i don't remember and don't have phpStorm open atm to look)